File: app/src/main/java/me/ycdev/android/devtools/root/cmd/AppsKillerCmd.java
package me.ycdev.android.devtools.root.cmd;

import android.content.Context;

import java.util.List;

import me.ycdev.android.devtools.root.RootCommandBuilder;
import timber.log.Timber;

public class AppsKillerCmd extends RootCmdBase {
    private static final String TAG = "AppsKillerCmd";

    private List<String> mPkgList;

    public AppsKillerCmd(Context cxt) {
        super(cxt);
    }

    public void setPackagesToKill(List<String> pkgList) {
        mPkgList = pkgList;
    }

    @Override
    public void run() {
        if (mPkgList == null || mPkgList.size() == 0) {
            Timber.tag(TAG).w("no apps to kill");
            return;
        }

        String[] cmds = RootCommandBuilder.forceStopPackage(mContext, mPkgList);
        runSuCommand(cmds, "u:r:system_app:s0");
    }
}

File: app/src/main/java/me/ycdev/android/devtools/root/cmd/AppsKillerCmd.kt
package me.ycdev.android.devtools.root.cmd

import android.content.Context
import me.ycdev.android.devtools.root.RootCommandBuilder
import timber.log.Timber

class AppsKillerCmd(cxt: Context, private val pkgList: List<String>) : RootCmdBase(cxt) {
    override fun run() {
        if (pkgList.isEmpty()) {
            Timber.tag(TAG).w("no apps to kill")
            return
        }
        val cmds = RootCommandBuilder.forceStopPackage(context, pkgList)
        RootCmdBase.Companion.runSuCommand(cmds, "u:r:system_app:s0")
    }

    companion object {
        private const val TAG = "AppsKillerCmd"
    }
}
